src/state/slices/opportunitiesSlice/selectors.test.ts

```diff
@@ -0,0 +1,272 @@
+import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { mockStore } from 'test/mocks/store'
+
+import { foxEthPair } from './constants'
+import type { UserStakingId } from './opportunitiesSlice'
+import { initialState } from './opportunitiesSlice'
+import {
+  selectAggregatedUserStakingOpportunityByStakingId,
+  selectHighestBalanceAccountIdIdByLpId,
+  selectHighestBalanceAccountIdIdByStakingId,
+  selectLpOpportunityIdsByAccountId,
+  selectStakingOpportunityIdsByAccountId,
+  selectUserStakingOpportunitiesByStakingId,
+  selectUserStakingOpportunityByUserStakingId,
+} from './selectors'
+import { serializeUserStakingId } from './utils'
+
+describe('opportunitiesSlice selectors', () => {
+  const baseState = {
+    ...mockStore,
+    opportunities: initialState,
+  }
+  describe('selects ID/s', () => {
+    const accountBalances = {
+      byId: {
+        'eip155:1:0xgomes': {
+          'eip155:1:0xLpContractOne': '1337',
+        },
+        'eip155:1:0xfauxmes': {
+          'eip155:1:0xLpContractOne': '424242',
+        },
+      },
+      ids: ['eip155:1:0xgomes', 'eip155:1:fauxmes'],
+    }
+    const lp = {
+      ...initialState.lp,
+      byAccountId: {
+        'eip155:1:0xgomes': ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+        'eip155:1:fauxmes': ['eip155:1:0xLpContractOne'],
+      },
+      ids: ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+    }
+    const staking = {
+      ...initialState.staking,
+      byAccountId: {
+        'eip155:1:0xgomes': ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+        'eip155:1:fauxmes': ['eip155:1:0xLpContractOne'],
+      },
+      ids: ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+    }
+    const userStaking = {
+      ...initialState.staking,
+      ids: [
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractTwo'),
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractOne'),
+        serializeUserStakingId('eip155:1:0xfauxmes', 'eip155:1:0xLpContractOne'),
+      ] as UserStakingId[],
```

Comment:
unnecessary cast?

Reply:
Nice yeah, the util will now give us the proper return value - d2d2ba4b5b827a0097b7cd4b62f93732218d4df7